src/scimlfunctions.jl

```diff
@@ -4199,7 +4199,7 @@ IntervalNonlinearFunction(f::IntervalNonlinearFunction; kwargs...) = f
 struct NoAD <: AbstractADType end
 
 (f::OptimizationFunction)(args...) = f.f(args...)
-OptimizationFunction(args...; kwargs...) = OptimizationFunction{true}(args...; kwargs...)
+OptimizationFunction(f, args...; kwargs...) = OptimizationFunction{isinplace(f, 3)}(f, args...; kwargs...)
```

Comment:
no, OptimizationFunction always has a scalar return so its iip is determined not by `f` but by the other functions such as the jac and cons

Reply:
fair enough. Will revert.